lib/engine/game/g_1871/step/dividend.rb

```diff
@@ -31,8 +31,17 @@ def share_price_change(_entity, revenue)
           end
 
           def payout_per_share(entity, revenue)
+            share_count =
+              if entity == @game.peir
+                # entity.total_shares doesn't always work correctly for PEIR.
+                # If there are three shares then each is a 33% share, and
+                # total_shares returns 3.030303...
+                @game.peir_shares.size
+              else
+                entity.total_shares
+              end
```

Comment:
Consider a ternary operator here.

Reply:
I had considered that, but the ternary operator made the line was too long for my tastes. But I've thought of a much cleaner implementation since…